python/mpt/src/mpt/trie_diff.py

```diff
@@ -0,0 +1,388 @@
+                # Right is prefix of left
+                elif l_node.key_segment.startswith(r_node.key_segment):
+                    # Compare the right node's value with the left node shortened by right key
+                    l_node.key_segment = l_node.key_segment[len(r_node.key_segment) :]
+                    self._compute_diff(
+                        l_node,
+                        r_node.subnode,
+                        path + r_node.key_segment,
+                        process_leaf_diff,
+                    )
+                # Left is prefix of right
+                elif r_node.key_segment.startswith(l_node.key_segment):
+                    # Compare the left node's value with the right node shortened by left key
+                    r_node.key_segment = r_node.key_segment[len(l_node.key_segment) :]
+                    self._compute_diff(
+                        l_node.subnode,
+                        r_node,
+                        path + l_node.key_segment,
+                        process_leaf_diff,
+                    )
+                # Both are different -> Look for diffs in both sub-trees
+                else:
+                    self._compute_diff(
+                        l_node.subnode,
+                        None,
+                        path + l_node.key_segment,
+                        process_leaf_diff,
+                    )
+                    self._compute_diff(
+                        None,
+                        r_node.subnode,
+                        path + r_node.key_segment,
+                        process_leaf_diff,
+                    )
+
+            case (ExtensionNode(), BranchNode()):
+                # Match on the corresponding nibble of the extension key segment
+                for i in range(0, 16):
+                    nibble = i.to_bytes(1, "big")
+                    if l_node.key_segment[0] == nibble:
+                        # Remove the nibble from the extension key segment
+                        l_node.key_segment = l_node.key_segment[1:]
+                        self._compute_diff(
+                            l_node.subnode,
+                            r_node.subnodes[i],
+                            path + nibble,
+                            process_leaf_diff,
+                        )
+                    else:
+                        # Look for diffs in other branches
+                        self._compute_diff(
+                            l_node.subnode,
+                            r_node.subnodes[i],
+                            path + nibble,
+                            process_leaf_diff,
+                        )
+
+            case (BranchNode(), None):
+                # Look for diffs in all branches of the left sub-tree
+                for i in range(0, 16):
+                    self._compute_diff(
+                        l_node.subnodes[i],
+                        None,
+                        path + i.to_bytes(1, "big"),
```

Comment:
```suggestion
                        path + bytes([i])
```
nit btw, if you prefere i.to_bytes, ok for me